From 6ce41498d8e7dfa1fade8cc74957e2509cd097ca Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:39 -0400
Subject: [fs] nfsd: remove unused function nfsd_read_file

Message-id: <1403560314-32525-158-git-send-email-bfields@redhat.com>
Patchwork-id: 84542
O-Subject: [Patch RHEL7 157/172] nfsd: remove unused function nfsd_read_file
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Christoph Hellwig <hch@lst.de>

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: b52bd7bccca57b2bbe837d14a0bf3e45279459a8
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/vfs.c b/fs/nfsd/vfs.c
index 10f2954..87d44fb 100644
--- a/fs/nfsd/vfs.c
+++ b/fs/nfsd/vfs.c
@@ -1177,26 +1177,6 @@ __be32 nfsd_read(struct svc_rqst *rqstp, struct svc_fh *fhp,
  return err;
 }
 
-/* As above, but use the provided file descriptor. */
-__be32
-nfsd_read_file(struct svc_rqst *rqstp, struct svc_fh *fhp, struct file *file,
-  loff_t offset, struct kvec *vec, int vlen,
-  unsigned long *count)
-{
- __be32  err;
-
- if (file) {
-  err = nfsd_permission(rqstp, fhp->fh_export, fhp->fh_dentry,
-    NFSD_MAY_READ|NFSD_MAY_OWNER_OVERRIDE);
-  if (err)
-   goto out;
-  err = nfsd_vfs_read(rqstp, file, offset, vec, vlen, count);
- } else /* Note file may still be NULL in NFSv4 special stateid case: */
-  err = nfsd_read(rqstp, fhp, offset, vec, vlen, count);
-out:
- return err;
-}
-
 /*
  * Write data to a file.
  * The stable flag requests synchronous writes.
diff --git a/fs/nfsd/vfs.h b/fs/nfsd/vfs.h
index 3a1c1e3..e5b35e5 100644
--- a/fs/nfsd/vfs.h
+++ b/fs/nfsd/vfs.h
@@ -83,8 +83,6 @@ int  nfsd_readv(struct file *, loff_t, struct kvec *, int,
     unsigned long *);
 __be32   nfsd_read(struct svc_rqst *, struct svc_fh *,
     loff_t, struct kvec *, int, unsigned long *);
-__be32   nfsd_read_file(struct svc_rqst *, struct svc_fh *, struct file *,
-    loff_t, struct kvec *, int, unsigned long *);
 __be32   nfsd_write(struct svc_rqst *, struct svc_fh *,struct file *,
     loff_t, struct kvec *,int, unsigned long *, int *);
 __be32  nfsd_readlink(struct svc_rqst *, struct svc_fh *,
-- 
1.7.1